airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/split_secrets/JsonSecretsProcessor.java

```diff
@@ -212,6 +217,52 @@ protected SecretKeys getAllSecretKeys(final JsonNode schema) {
     return new SecretKeys(fieldSecretKeys, arraySecretKeys);
   }
 
+  /**
+   * This method extract all the json path of all secret in a json schema.
+   * The path are not the ones that match the path in the schema but will be matching the ones of the object described by the schema.
+   *
+   * @param currentNode - The node currently being process
+   * @param currentPath - The current path of the node
+   * @return
+   */
+  protected Set<String> getAllSecretPathsRec(final JsonNode currentNode, final String currentPath) {
```

Comment:
it feels like we're adding a 3rd jsonschema tree traversal to our codebase here. that makes me nervous. i have a generic implementation that i've been working on to replace these duplicative implementations. will share this afternoon.